src/revivalpmmp/pureentities/data/Color.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+
+namespace revivalpmmp\pureentities\data;
+
+class Color {
+
+    const COLOR_DYE_BLACK = 0;//dye colors
+    const COLOR_DYE_RED = 1;
+    const COLOR_DYE_GREEN = 2;
+    const COLOR_DYE_BROWN = 3;
+    const COLOR_DYE_BLUE = 4;
+    const COLOR_DYE_PURPLE = 5;
+    const COLOR_DYE_CYAN = 6;
+    const COLOR_DYE_LIGHT_GRAY = 7;
+    const COLOR_DYE_GRAY = 8;
+    const COLOR_DYE_PINK = 9;
+    const COLOR_DYE_LIME = 10;
+    const COLOR_DYE_YELLOW = 11;
+    const COLOR_DYE_LIGHT_BLUE = 12;
+    const COLOR_DYE_MAGENTA = 13;
+    const COLOR_DYE_ORANGE = 14;
+    const COLOR_DYE_WHITE = 15;
+
+
+    const WHITE = 0;
+    const ORANGE = 1;
+    const MAGENTA = 2;
+    const LIGHT_BLUE = 3;
+    const YELLOW = 4;
+    const LIME = 5;
+    const PINK = 6;
+    const GRAY = 7;
+    const LIGHT_GRAY = 8;
+    const CYAN = 9;
+    const PURPLE = 10;
+    const BLUE = 11;
+    const BROWN = 12;
+    const GREEN = 13;
+    const RED = 14;
+    const BLACK = 15;
+
+    /** @var \SplFixedArray */
+    public static $dyeColors = null;
+    private static $convertedColors = null;
+    private $red = 0;
+    private $green = 0;
+    private $blue = 0;
+
+    public function __construct($r, $g, $b) {
+        $this->red = $r;
+        $this->green = $g;
+        $this->blue = $b;
+    }
+
+    public static function init() {
+        if (self::$dyeColors === null) {
+            self::$dyeColors = new \SplFixedArray(16); //What's the point of making a 256-long array for 16 objects?
+            self::$dyeColors[self::COLOR_DYE_BLACK] = Color::getRGB(30, 27, 27);
+            self::$dyeColors[self::COLOR_DYE_RED] = Color::getRGB(179, 49, 44);
+            self::$dyeColors[self::COLOR_DYE_GREEN] = Color::getRGB(61, 81, 26);
+            self::$dyeColors[self::COLOR_DYE_BROWN] = Color::getRGB(81, 48, 26);
+            self::$dyeColors[self::COLOR_DYE_BLUE] = Color::getRGB(37, 49, 146);
+            self::$dyeColors[self::COLOR_DYE_PURPLE] = Color::getRGB(123, 47, 190);
+            self::$dyeColors[self::COLOR_DYE_CYAN] = Color::getRGB(40, 118, 151);
+            self::$dyeColors[self::COLOR_DYE_LIGHT_GRAY] = Color::getRGB(153, 153, 153);
+            self::$dyeColors[self::COLOR_DYE_GRAY] = Color::getRGB(67, 67, 67);
+            self::$dyeColors[self::COLOR_DYE_PINK] = Color::getRGB(216, 129, 152);
+            self::$dyeColors[self::COLOR_DYE_LIME] = Color::getRGB(65, 205, 52);
+            self::$dyeColors[self::COLOR_DYE_YELLOW] = Color::getRGB(222, 207, 42);
+            self::$dyeColors[self::COLOR_DYE_LIGHT_BLUE] = Color::getRGB(102, 137, 211);
+            self::$dyeColors[self::COLOR_DYE_MAGENTA] = Color::getRGB(195, 84, 205);
+            self::$dyeColors[self::COLOR_DYE_ORANGE] = Color::getRGB(235, 136, 68);
+            self::$dyeColors[self::COLOR_DYE_WHITE] = Color::getRGB(240, 240, 240);
+        }
+
+        if (self::$convertedColors === null) {
```

Comment:
I'm not a PHP programmer, but a JAVA programmer since A LOT of years. PHP isn't my favorite programming language. So i assume this is a PHP built-in function? Don't know much about PHP built-in functions. But you can contribute :smile: 

Apart from that: this class is copied from Tesseract - as PocketMine-MP doesn't have it. But in the end i saw, that it's only used by PureEntities to convert dye colors to entity colors. Maybe i will remove all the other stuff from this class ...

Anyway - thanks a lot for reviewing!